text/shared/src/main/scala/fs2/data/text/render/package.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024 fs2-data Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.data.text
+
+import fs2.Pipe
+import fs2.data.text.render.internal.StreamPrinter
+
+package object render {
+
+  def pretty[F[_], Event](width: Int = 100, indent: Int = 2)(implicit
```

Comment:
If I wanted to indent with tabs instead of spaces, I couldn't do, right? Is there maybe an easy generalisation of `indent` that allows for this?

Reply:
Well... How big is a tab? The pretty printing algorithm based on Wadler's (basically what I implemented here) usually tab with space and monospace fonts.

I know of other algorithms allowing for more flexible layouts (like variable width font, and maybe tabs), but TBH this did not seem worth implementing in this case.

Reply:
I briefly thought about something along the lines of
```scala
enum Indentation {
  case Spaces(amount: Int)
  case Tab(amount: Int, width: Int)
  case Custom(amount: Int, width: Int /* = value.length */, value: String)
}
```
but then there'd also be a fair argument saying it should be extendable, so an open `trait` with some methods for the relevant properties and then things escalate quickly. So yes, limiting to spaces & monospace fonts seems reasonable.